client.go

```diff
@@ -346,13 +346,13 @@ func (c *Client) Write(path string, data []byte, _ os.FileMode) error {
 		return nil
 
 	case 409:
-		if i := strings.LastIndex(path, "/"); i > -1 {
-			if err := c.MkdirAll(path[0:i+1], 0755); err == nil {
-				s = c.put(path, bytes.NewReader(data))
-				if s == 200 || s == 201 || s == 204 {
-					return nil
-				}
+		if err := c.createParentCollection(path); err == nil {
+			s = c.put(path, bytes.NewReader(data))
```

Comment:
use an early return on error or drop the return error case, fall through

Reply:
done in [eca9684](https://github.com/studio-b12/gowebdav/pull/22/commits/eca9684e7c84f8f5a593c528877a8a29d2a2b21a)